### PR TITLE
[trace][feature] Local cache for workspace kind & support `trace.provider=local`

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

**Local cache for Azure ML workspace/AI project type**

Local cache resource type, so that we don't need to request Azure everytime. There is a timeout 1 day, so that we can avoid missing the Cosmos status in the future.

**Support `trace.provider=local`**

Enable local only trace feature.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-devkit/promptflow/_sdk/_tracing_utils.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+from promptflow._sdk._constants import HOME_PROMPT_FLOW_DIR, AzureMLWorkspaceTriad
+from promptflow._sdk._utils import json_load
+from promptflow._utils.logger_utils import get_cli_sdk_logger
+from promptflow.core._errors import MissingRequiredPackage
+
+_logger = get_cli_sdk_logger()
+
+PF_DIR_TRACING = "tracing"
+WORKSPACE_KIND_LOCAL_CACHE_EXPIRE_DAYS = 1
```
Comment (crazygao) [inline src/promptflow-devkit/promptflow/_sdk/_tracing_utils.py:19]: Move these constants inside the class?

File: src/promptflow-devkit/tests/sdk_cli_test/unittests/test_trace.py [hunk 2]
```diff
@@ -211,3 +213,67 @@ def _validate_invalid_params(kwargs: Dict):
...
+            f.write(json.dumps(cache))
+        ws_local_cache = WorkspaceKindLocalCache(
+            subscription_id=mock_sub, resource_group_name=mock_rg, workspace_name=mock_ws
+        )
+        assert ws_local_cache.is_cache_exists is True
+        assert not ws_local_cache.is_expired
+        assert ws_local_cache.get_kind() == mock_kind
+
+    def test_expired_cache(self):
+        mock_sub, mock_rg, mock_ws = str(uuid.uuid4()), str(uuid.uuid4()), str(uuid.uuid4())
```
Comment (crazygao) [inline src/promptflow-devkit/tests/sdk_cli_test/unittests/test_trace.py:256]: change the word mock to something else...
In these tests.

File: src/promptflow-devkit/promptflow/_sdk/_tracing_utils.py (hunk 1 above)
Comment (zhengfeiwang) [reply src/promptflow-devkit/promptflow/_sdk/_tracing_utils.py:19]: make sense, let me update this

File: src/promptflow-devkit/tests/sdk_cli_test/unittests/test_trace.py (hunk 2 above)
Comment (zhengfeiwang) [reply src/promptflow-devkit/tests/sdk_cli_test/unittests/test_trace.py]: updated to remove these "mock" usage

File: src/promptflow-devkit/promptflow/_sdk/_tracing_utils.py (hunk 1 above)
Comment (zhengfeiwang) [reply src/promptflow-devkit/promptflow/_sdk/_tracing_utils.py]: updated

**Review by crazygao — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**